app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,29 @@
+module Katello
+  class RepositoryTypeManager
+    @repository_types = {}
+    class << self
+      attr_reader :repository_types
+      private :new
+      # Plugin constructor
+      def register(id, &block)
+        repository_type = ::Katello::RepositoryType.new(id)
+        repository_type.instance_eval(&block)
+        repository_types[id.to_s] = repository_type
+      end
+
+      def creatable_repository_types
+        repository_types.select do |repo_type, _|
+          creatable_by_user?(repo_type)
+        end
+      end
+
+      def creatable_by_user?(repository_type)
+        find(repository_type).allow_creation_by_user
+      end
+
+      def find(repository_type)
+        repository_types[repository_type.to_s]
+      end
+    end
+  end
+end
```

Comment:
I didn't like the  self.method especially when every method in this class is self.x ...